_posts/2016-04-04-code-smells-part-II.md

```diff
@@ -0,0 +1,91 @@
+   Similar to the above one, in this case, you should make use of one of the patterns: [State](https://en.wikipedia.org/wiki/State_pattern) or [Strategy](https://en.wikipedia.org/wiki/Strategy_pattern).
+
+So when to use one or the other? If the **_Type Code_** does not change the behaviour of a class you can use the **_Subclasses_** technique. Separating each behaviour in it's appropriated subclass will enforce the [Single Responsibility Principle](https://en.wikipedia.org/wiki/Single_responsibility_principle) and make the code more readable in general. And if you need to add another case, you just need to add a new class to your code without having to modify any other code. So you apply the [Open/Close Principle](https://en.wikipedia.org/wiki/Open/closed_principle).
+You should use the Strategy approach when the **_Type Code_** affects the behaviour of your classes. If you're changing the state of the class, fields and many other actions then you should use the [State Pattern](https://en.wikipedia.org/wiki/State_pattern). if only affects how you select a behaviour of the class then the [Strategy Pattern](https://en.wikipedia.org/wiki/Strategy_pattern) is your man.
+
+#### Temporary Field
+
+This case occurs when we are calculating some big algorithm that needs several input variables. Creating these fields in the class has no value most of the times because they are just used for this specific calculation. And this can be dangerous too because you have to be sure you reinitialize then before you start the next computation. 
+Here the best refactoring technique is to use **_Replace Method with Method Object_**, wich will extract the method into a separate class. Then you can split the method into several methods within the same class. 
+
+
+#### Refused Bequest
+
+This code smell is a little tricky to detect because this happens when a subclass doesn't use all the behaviour of its parent class. So it's as if the subclass "refuses" some behaviour("bequest") of its parent class. 
+
+In this case, if it makes no sense to continue to have this inheritance, the best refactoring technique is to use **_Delegation_**: we can get rid of the inheritance by creating a field in our subclass and putting in there the parent class. This way every time you need the methods from the parent class just delegate then to this new object. 
+
+When the inheritance is the correct thing to do, then move all unnecessary fields and methods from the subclass. Extract all methods and fields from the subclass and parent class and put then in a new class. Make this new class the SuperClass, from whom the subclass and parent class should inherit. This technique is called **_Extract Superclass_**.
+
+
+#### Alternative Classes with Different Interfaces
+
+Hmm, this case makes me think of "lack of communication" between members of the same team because this happens when we have two classes that make basically the same thing but have different names in their methods. 
```

Comment:
different names in their methods =>...different names for their methods ... is this what you want to say.